gordo_components/client/client.py

```diff
@@ -121,6 +126,39 @@ def __init__(
         self.target = target
         self.machines = self.get_machines()
 
+    @property
+    def revision(self):
+        """
+        Revision being used against the server
+        """
+        return self._revision
+
+    @revision.setter
+    def revision(self, revision: str):
+        """
+        Verifies the server can satisfy this revision
+        """
+        resp = self.session.get(
+            f"{self.base_url}/gordo/v0/{self.project_name}/revisions"
+        )
+        if resp.ok:
+            if revision:
```

Comment:
This looks like it would always evaluate since it is a required arg to this method, but not familiar with property setting decorators.

Reply:
Ah, it of coarse wouldn't if some null thing was passed, but in that case, should it maybe just error rather than setting the revision to latest?

Reply:
ya.. I'm inclined to agree, a bit too magical that setting `self.revision = None` will result in finding the latest; at best passing 'latest' or method to update to latest. I'd guess I'll opt for the former.